### PR TITLE
Update wpcs 3.0.0

### PR DESCRIPTION
**Pull request by dingo-d**
# Description

Still a WIP due to an upstream bug: https://github.com/WordPress/WordPress-Coding-Standards/pull/2370

This update will sync the CS with the latest WPCS 3.0.0, and do some quality of life improvements.

To do:

- [x] Update changelog with all the changes
- [x] Update readme with all the changes
- [x] Update CI checks
- [x] Test that everything passes
- [ ] Test the new public property in the ComponentsEscape sniff
- [ ] Test the new branch (once all bugs pass) on libs/projects (shouldn't block PHP 8 installation anymore)
- [ ] Tag v2 of the Eightshift Coding Standards

### CODE REVIEW

File: Eightshift/Sniffs/Security/ComponentsEscapeSniff.php [hunk 1]
```diff
@@ -42,6 +42,16 @@ class ComponentsEscapeSniff extends EscapeOutputSniff
 	 */
 	public $overriddenClass = '';
 
+	/**
+	 * List of allowed methods that won't trigger the EscapeOutput error.
+	 *
+	 * @var true[]
```
Comment (dingo-d) [inline Eightshift/Sniffs/Security/ComponentsEscapeSniff.php:48]: Note to self: Update this.

And add phpstan as a dev dependency for better type safety.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        php: ['7.2', '7.3', '7.4']
+        php: ['7.4', '8.0', '8.1']
```
Comment (dingo-d) [inline .github/workflows/lint.yml:24]: Note to self: Check the previously closed PRs for other updates (checkout v3 etc.)

**Comment by dingo-d** [comment]
To do: rebase useless commits into one.
Currently, the CI fails on WPCS 3.0.0, but that is expected as the fix is still not merged, and will be released in v3.1.0.

**Comment by dingo-d** [comment]
@goranalkovic-infinum @iruzevic if you agree, I can rebase this pr and we can merge it and tag 2.0.0-beta, because the release will depend on the upstream fix in the wpcs that will be released in 3.1.0 (don't know when this will happen tbh).

Then you can release new libs version with this beta CS version so that you're not blocked on PHP8+.

Oh, and also you can pass this to the management: 
https://opencollective.com/thewpcc/contribute/wp-php-63406

😁

**Review by goranalkovic-infinum — APPROVED** [review]

**Review by piqusy — APPROVED** [review]

**Pull request merged**